conans/client/graph/range_resolver.py

```diff
@@ -143,24 +143,24 @@ def _resolve_local(self, search_ref, version_range):
         if local_found:
             return self._resolve_version(version_range, local_found)
 
-    def _search_remotes(self, pattern, remotes):
-        remote = remotes.selected
-        if remote:
-            search_result = self._remote_manager.search_recipes(remote, pattern, ignorecase=False)
-            return search_result, remote.name
-
+    def _search_remotes(self, search_ref, remotes):
         for remote in remotes.values():
-            search_result = self._remote_manager.search_recipes(remote, pattern, ignorecase=False)
-            if search_result:
-                return search_result, remote.name
+            if not remotes.selected or remote == remotes.selected:
+                search_result = self._remote_manager.search_recipes(remote, search_ref.name,
+                                                                    ignorecase=False)
+                search_result = [ref for ref in search_result
+                                 if ref.user == search_ref.user and
+                                 ref.channel == search_ref.channel]
+                if search_result:
+                    return search_result, remote.name
         return None, None
 
     def _resolve_remote(self, search_ref, version_range, remotes):
         # We should use ignorecase=False, we want the exact case!
         found_refs, remote_name = self._cached_remote_found.get(search_ref, (None, None))
         if found_refs is None:
             # Searching for just the name is much faster in remotes like Artifactory
-            found_refs, remote_name = self._search_remotes(search_ref.name, remotes)
+            found_refs, remote_name = self._search_remotes(search_ref, remotes)
             if found_refs:
                 found_refs = [r for r in found_refs
                               if r.user == search_ref.user and r.channel == search_ref.channel]
```

Comment:
As you are filtering by `user/channel` inside `self._search_remotes` these lines are no longer needed...